app/src/main/kotlin/xyz/duckee/android/MainActivity.kt

```diff
@@ -109,20 +107,47 @@ class MainActivity : ComponentActivity() {
                 }
             }
         }
+
+        // Handle deeplink when the app is opened from a deeplink
+        handleDeeplink(intent)
     }
 
     override fun onResume() {
         super.onResume()
 
-        Firebase.dynamicLinks
-            .getDynamicLink(intent)
-            .addOnSuccessListener(this) { pendingDynamicLinkData: PendingDynamicLinkData? ->
-                pendingDynamicLinkData?.link?.let { deepLink ->
-                    // Handle Deeplink with Jetpack NavigationComponent
-                    navigationController?.navigate(deepLink)
-                }
+        // Handle deferred deeplink
+        Airbridge.handleDeferredDeeplink { uri ->
+            uri?.let {
+                // Handle the deferred deeplink URI
+                // For example, navigate to a specific screen based on the URI
+                handleDeeplinkUri(it)
+            }
+        }
+    }
+
+    override fun onNewIntent(intent: Intent?) {
+        super.onNewIntent(intent)
+        intent?.let { handleDeeplink(it) }
+    }
+
+    private fun handleDeeplink(intent: Intent) {
+        Airbridge.handleDeeplink(intent) { uri ->
+            // Handle the deeplink URI
+            // For example, navigate to a specific screen based on the URI
+            handleDeeplinkUri(uri)
+        }
+    }
```

Comment:
Added `handleDeeplink()` method to process deeplinks using Airbridge SDK. This method replaces the Firebase Dynamic Links handling and provides a centralized place for deeplink processing.